test/run_test.go

```diff
@@ -0,0 +1,187 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/flanksource/kommons"
+	"github.com/flanksource/template-operator-library/pkg/health"
+	"github.com/mitchellh/go-homedir"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type TestInstance struct {
+	CRD         string
+	Template    string
+	Fixture     string
+	Test        func(client *kommons.Client, namespace string) (bytes.Buffer, error)
+	ReadyChecks []ReadyCheck
+}
+
+type ReadyCheck struct {
+	Name                 string
+	Resource             string
+	ResourceReadyTimeout time.Duration
+	WaitFN               kommons.WaitFN
+}
+
+var TestSpace []TestInstance
+
+func TestMain(m *testing.M) {
+	TestSpace = []TestInstance{
+		{
+			CRD:      "../config/crd/db/db.flanksource.com_elasticsearchdbs.yaml",
+			Template: "../config/templates/elasticsearch-db.yaml",
+			Fixture:  "fixtures/elasticsearch-db.yaml",
+			Test:     health.ElasticsearchCheck,
+			ReadyChecks: []ReadyCheck{
+				{
+					Resource:             "Elasticsearch",
+					ResourceReadyTimeout: 12 * time.Minute,
+					WaitFN:               health.IsElasticReady,
+				},
+			},
```

Comment:
We shouldn't need to explicitly specify the test/ready functions.

they should all use kommons.WaitForResourceRecursive() walk the ownership tree and checking IsReady on each resource.